akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExample3b.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.PersistentBehavior3
+
+/*
+API experiment with abstract class.
+- onCommand (commandHandler) and onEvent (eventHandler) are methods
+- see also alignment with scaladsl.MutableBehavior
+- using PersistentBehavior3
+ */
+
+object AccountExample3b {
```

Comment:
I think this is my preference.

It is most similar to our existing approaches. This reminds me a lot of Lagom Persistent Entities, and it is also fairly similar to Akka Persistence (Untyped). That familiarity will make transitioning existing users over to the new approach easier.

One change I would make is that instead of the enclosing class being "PersistentBehavior" I would make it something like "PersistentEntity" or "PersistentActor". Again, this increases the similarity to existing approaches.

Reply:
@WadeWaldron,

In Akka Typed, everything is a Behavior. Which basically comes down to define state and functions that will be run inside an actor. 

One thing is a `PersistentBehavior` another thing is a sharded `PersistentBehavior`. When we shard it, it becomes an `Entity`. 

You can shard any actor. You go from an `ActorRef[T]` (just an actor) to an `Entity[T]` (a sharded actor). 

You can make an `ActorRef[T]` from a `Behavior` and it can also be a `PersistentBehavior`.

Eventually, we may want (or need) to introduce a `PersistentEntity`. That will mean, an Actor, built from a `PersistentBehavior`, but sharded (wrapped by `Entity[T]`). 

The term `PersistentEntity` should be reserved for now.

Reply:
I understand Wade's points, but I agree that it is more consistent to call it PersistentBehavior. However, I think it is very important that we take a serious look at providing a PersistentEntity that guides the user onto the right path in using Cluster Sharding.

Reply:
Sorry for bikeshedding this, but, isn't "PersistentBehavior" a misnomer?
Since it is not about the behavior being persistent, but the state.

Reply:
So perhaps "PersistingBehavior"?

Reply:
> So perhaps "PersistingBehavior"?

I can live with that color of the shed ;-)

Reply:
That would be how to send the commands via Cluster Sharding. https://doc.akka.io/docs/akka/snapshot/typed/cluster-sharding.html

I can add that to this PR so we see the full picture.

Reply:
That would be wonderful @patriknw ..

Reply:
@debasishg I have added AccountExample5bWithSharding
https://github.com/akka/akka/pull/25503/files#diff-a191dbf294d5a19cdcf75c9aad17755f

Reply:
@jboner I think maybe what I am looking for is the same as what you are looking for. If there was a "PersistentEntity" that guided people to sharding, and looked a bit like this code, then I would be pretty happy.